benchmarks/src/environment/snake.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Dict, Any
+
+import dataclasses
+import logging
+import subprocess
+import os
+
+from pathlib import Path
+from . import Environment, EnvironmentDescriptor
+
+
+@dataclasses.dataclass(frozen=True)
+class SnakeClusterInfo:
+    workdir: Path
+
+
+class SnakeEnvironmentDescriptor(EnvironmentDescriptor):
+    def create_environment(self, workdir: Path) -> Environment:
+        info = SnakeClusterInfo(workdir)
+        return SnakeEnvironment(info)
+
+    def name(self) -> str:
+        return "snake"
+
+    def parameters(self) -> Dict[str, Any]:
+        return {}
+
+    def metadata(self) -> Dict[str, Any]:
+        return {}
+
+
+class SnakeEnvironment(Environment):
+    def __init__(self, info: SnakeClusterInfo):
+        self.info = info
+        self.snakefile = "Snakefile"
+        self.state = "initial"
```

Comment:
The state checking will be needed in most environments, I added a mixin class for it (`EnvStateManager`) in `main`. Please use it here.